lib/firebase_database.dart

```diff
@@ -9,66 +9,75 @@ final firebaseUrl =
-    final response = await http.patch(
-      uri,
-      body: body,
-    );
+  static Future<void> saveGooglers(List googlers) async {
+    final uri = Uri.parse('$firebaseUrl.json');
+    final response =
+        await http.patch(uri, body: jsonEncode({'googlers': googlers}));
     if (response.statusCode != 200) {
-      throw Exception(
-          'Error setting last_updated for ${slug.fullName} in $type $body: ${response.body}');
+      throw Exception('Error adding Googlers ${response.body}');
     }
   }
 
-  Future<void> deleteAllData() async {
-    final uri = Uri.parse('$firebaseUrl${type.name}/data/${slug.toUrl()}.json');
-    final response = await http.delete(uri);
+  static Future<void> setLastUpdated(RepositorySlug slug) async {
+    final uri = Uri.parse('${firebaseUrl}last_updated.json');
+    final response = await http.patch(uri,
+        body:
+            jsonEncode({slug.toUrl(): DateTime.now().millisecondsSinceEpoch}));
     if (response.statusCode != 200) {
-      throw Exception(
-          'Error delete data for ${slug.fullName} in $type: ${response.body}');
+      throw Exception('Error adding Googlers ${response.body}');
     }
   }
 
-  Future<void> addData(String data) async {
-    final uri = Uri.parse('$firebaseUrl${type.name}/data/${slug.toUrl()}.json');
-    final response = await http.patch(uri, body: data);
+  static Future<Map<RepositorySlug, DateTime?>> getLastUpdated() async {
+    final uri = Uri.parse('${firebaseUrl}last_updated.json');
+    final response = await http.get(uri);
     if (response.statusCode != 200) {
-      throw Exception(
-          'Error delete data for ${slug.fullName} in $type: ${response.body}');
+      throw Exception('Error adding Googlers ${response.body}');
     }
+    final map = (jsonDecode(response.body) ?? {}) as Map<String, dynamic>;
+    return map.map((key, value) => MapEntry(
+        RepositorySlugExtension.fromUrl(key),
+        DateTime.fromMillisecondsSinceEpoch(value)));
   }
 
-  static Future<void> saveGooglers(String googlers) async {
-    final uri = Uri.parse('$firebaseUrl.json');
-    final response =
-        await http.patch(uri, body: jsonEncode({'googlers': googlers}));
-    if (response.statusCode != 200) {
-      throw Exception('Error adding Googlers ${response.body}');
+  static List<T> extractDataFrom<T>(
+    Map<String, dynamic> idsToData,
+    T Function(Map<String, dynamic> json) fromJson,
+  ) {
+    final list = <T>[];
+    for (final idToData in idsToData.entries) {
+      // ignore: unused_local_variable
+      final id = idToData.key;
+      final data = fromJson(idToData.value);
+      list.add(data);
     }
+    return list;
   }
 }
 
-extension on RepositorySlug {
+extension RepositorySlugExtension on RepositorySlug {
   String toUrl() {
     final ownerClean = owner.replaceAll(r'.', r',');
     final nameClean = name.replaceAll(r'.', r',');
     return '$ownerClean:$nameClean';
   }
+
+  static RepositorySlug fromUrl(String url) {
```

Comment:
Interesting - I wasn't aware your could create a static extension method.

Reply:
You can't really - it will be a static method on the extension class itself, not the extended class. So you have to call it as `RepositorySlugExtension.fromUrl`...